### PR TITLE
Fix method `convertException()`

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Exception/ConvertException.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        if ($this->e instanceof Exception) {
+            return $this->e;
+        }
+
+        $message = $this->e->getMessage() . PHP_EOL . 'The SQL being executed was: ' . $this->rawSql;
+        $errorInfo = $this->e instanceof PDOException ? $this->e->errorInfo : null;
+        $exception = new Exception($message, $errorInfo, $this->e);
+
+        if (str_contains($message, 'SQLSTATE[23') || str_contains($message, 'ORA-00001: unique constraint')) {
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Exception/ConvertException.php:25]: fixed.

File: src/Exception/ConvertException.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public function run(): Exception
+    {
+        if ($this->e instanceof Exception) {
+            return $this->e;
+        }
+
+        $message = $this->e->getMessage() . PHP_EOL . 'The SQL being executed was: ' . $this->rawSql;
+        $errorInfo = $this->e instanceof PDOException ? $this->e->errorInfo : null;
+        $exception = new Exception($message, $errorInfo, $this->e);
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Exception/ConvertException.php:26]: done.

File: src/Exception/ConvertException.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        if (
+            str_contains($message, self::MSG_INTEGRITY_EXCEPTION_1)
+            || str_contains($message, self::MGS_INTEGRITY_EXCEPTION_2)
+        ) {
+            $exception = new IntegrityException($message, $errorInfo, $this->e);
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Exception/ConvertException.php:32]: done.

File: src/Exception/ConvertException.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        return $exception;
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Exception/ConvertException.php:35]: done.

**Review by darkdef — APPROVED** [review]

**Pull request merged**